v0/destinations/woopra/utils.js

```diff
@@ -0,0 +1,30 @@
+const { set } = require("lodash");
+const { getBrowserInfo } = require("../../util");
+/**
+ *
+ * @param {*} object
+ * concatenates only values of the object
+ * @returns the concatenated string
+ */
+const mergeObjectValues = object => {
+  let res;
+  Object.keys(object).forEach(v => {
+    if (object[v]) {
+      res.concat(object[v], " ");
+    }
+  });
+  return res;
+};
+
+/**
+ * it adds the field requiring some manipulation to the payload
+ * @param {*} message
+ * @param {*} payload
+ */
+const refinePayload = (message, payload) => {
+  const app = message.context?.app?.name + message.context?.app?.build;
+  const browser = mergeObjectValues(getBrowserInfo(message.context.userAgent));
+  set(payload, "app", app);
+  set(payload, "browser", browser);
```

Comment:
Don't modify payload here. Create a new variable , do the changes and return that variable here.
In caller function use the response return by `refinePayload`

Reply:
we can do it either wise but I opted for passing the payload as Param because otherwise we have to take another variable and then performing merging of two payloads which will again take space and time.